flow/compositor_context.cc

```diff
@@ -88,7 +88,7 @@ RasterStatus CompositorContext::ScopedFrame::Raster(
   // paints some raster cache.
   if (canvas()) {
     if (needs_save_layer) {
-      FML_LOG(INFO) << "Using SaveLayer to protect non-readback surface";
+      TRACE_EVENT0("flutter", "Canvas::saveLayer");
```

Comment:
The cost of the save layer is not just isolated to this scope. There will be significant time taken in the flush waiting on the GPU to service this workload. So, perhaps a more actionable metric is to have a counter that tracks the number of save layers?

Reply:
`TRACE_COUNTER` instead of `TRACE_EVENT` I mean.

Reply:
Counters won't show up in systracing. My idea here is that devtools could manually count these, and developers would be able to more easily see what callstacks were resulting in them. Then we'd want something like debugging the SKP that resulted and you could visually see what parts of the draw used a savelayer.

Reply:
(so yes, these trace events will be very small, but devtools will be able to scan for them and know they result in more expensive operations later)

Reply:
It's not just that the bulk of the work happens in a different phase of the pipeline (i.e. flush vs LayerTree.Paint), it's that even the saveLayer call itself doesn't really do the work. It remembers a few items you told it, then later the restore applies that work. So there are many reasons why this timed event isn't capturing any real work.

Why doesn't devtools scan for the counter events? They are just another [type of event](https://docs.google.com/document/d/1CvAClvFfyA5R-PhYUmn5OOQtYMH4h6I0nSsKchNAySU/preview#heading=h.msg3086636uq) in the Chrome Tracing.

Reply:
> Why doesn't devtools scan for the counter events?

Discussing this in person, it looks like devtools should be able to detect counter events. systrace is an issue though and I am investigating if counters are not supported by systrace or whether this is a limitation with our trace logger to systrace. In any case, it looks like our use case ought to be address by just tracing to the timeline.

Reply:
it would be great to switch this to a counter as these events are adding noise to the trace right now without clear user value.

Reply:
We didn't use counters because it wouldn't be seen by users. The action a user should take is to turn on debugTracePaintEnabled and remove or rewrite the render object adding this call.

Reply:
> ... without clear user value.

I don't remember the full context of this discussion---if these are trace events we explicitly added for the benefit of end-users, then we should certainly try to make them more useful---but in general, it's been our practice to add trace events as needed to assist Engine developers in zeroing in on issues, be they performance issues or otherwise. So (in general, maybe not for this specific case) if a trace event is too noisy, the corrective action would be to filter it before an end-user sees it.